From 2cda3a7e7dba72e7290fe0673b9d219e2b0d9682 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@intel.com>
Date: Thu, 10 Jan 2019 12:38:29 +0000
Subject: [PATCH 1/3] Provide a meaningful Makefile

This will help us to package up the extension for distribution.

Signed-off-by: Ikey Doherty <ikey.doherty@intel.com>
---
 Makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index 5d256bf..89f369d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,13 +1,13 @@
-all: schemas zip
+datadir ?= /usr/share
 
-0: phony
-	mkzero-gfxmonk -p impatience -p xdg gnome-shell-impatience.xml
+dummy:
 
-schemas: phony
-	cd impatience/schemas && glib-compile-schemas .
+install-data:
+	install -D -d -m 00755 $(DESTDIR)$(datadir)/gnome-shell/extensions/impatience@gfxmonk.net
+	install -m 00644 impatience/*.{js,json} $(DESTDIR)$(datadir)/gnome-shell/extensions/impatience@gfxmonk.net/.
+	install -D -d -m 00755 $(DESTDIR)$(datadir)/glib-2.0/schemas
+	install -m 00644 impatience/schemas/*.xml $(DESTDIR)$(datadir)/glib-2.0/schemas/.
 
-zip: phony
-	rm impatience@gfxmonk.net.zip
-	cd impatience && zip -r ../impatience@gfxmonk.net.zip *
+install: install-data
 
-.PHONY: phony
+.PHONY: dummy
-- 
2.20.0

